### PR TITLE
feat(security): Add Generate Registry Token API for secretstore client

### PR DESCRIPTION
**Pull request by jim-wang-intel**
Added vault's /consul/creds/<serviceKey> API  to generate consul token

Closes: #95

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [x ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

N/A: no API to generate consul token

Issue Number:  #95 

## What is the new behavior?
New management API GenerateRegistryToken is added on `SecretstoreClient` that can be used to generate consul token

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

## Other information

### CODE REVIEW

File: secrets/interfaces.go [hunk 1]
```diff
@@ -52,4 +52,13 @@ type SecretStoreClient interface {
...
 	RevokeToken(token string) error
+
+	// GenerateRegistryToken generates a new registry token based on the given serviceKey
+	// it requires a secret store token with the permission to generate the registry token
+	// the registry token is like a bearer token and is used to access the information from the registry
+	// like service's configuration in key/value store of the registry
+	// the generated token is unique every time
+	// caller should persist or cache the generated registry token, at least per runtime cycle, to reduce the number of
+	// tokens stored in registry server side and the number of calls to this API
+	GenerateRegistryToken(token, serviceKey string) (string, error)
```
Comment (lenny-intel) [inline secrets/interfaces.go:63]: Change to GenerateConsulToken

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/pkg/vault/management.go [hunk 2]
```diff
@@ -193,3 +198,62 @@ func (c *Client) CheckSecretEngineInstalled(token string, mountPoint string, eng
 
 	return false, nil
 }
+
+// GenerateRegistryToken generates a new registry token using serviceKey as role name to
+// call secretstore's consul/creds API
+// the serviceKey is used in the part of secretstore's URL as role name and should be accessible to the API
+func (c *Client) GenerateRegistryToken(mgmtToken, serviceKey string) (string, error) {
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply internal/pkg/vault/management.go:205]: ok, i'll change the naming

[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply internal/pkg/vault/management.go]: fix

File: internal/pkg/vault/management.go [hunk 3]
```diff
@@ -193,3 +198,62 @@ func (c *Client) CheckSecretEngineInstalled(token string, mountPoint string, eng
...
+	if len(mgmtToken) == 0 {
+		return emptyToken, pkg.NewErrSecretStore("missing mgmt token for generating registry token")
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply internal/pkg/vault/management.go]: fix

File: secrets/interfaces.go (hunk 1 above)
Comment (jim-wang-intel) [reply secrets/interfaces.go]: fix

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**